specification/dimensions/commitmentdiscountid.md

```diff
@@ -0,0 +1,32 @@
+# Commitment Discount ID
+
+A commitment-based discount is a contractual commitment for an amount of usage or spend, for a specified term, in exchange for a discounted unit price. Commitment-based discounts can cover either usage or cost of various resources such as virtual machines or databases. A provider can offer various pricing models, payment options, and term durations for commitment-based discounts.
+
+A Commitment Discount ID is the identifier assigned to a commitment-based discount by the provider. Commitment Discount ID is commonly used for scenarios like chargeback for commitments and savings per commitment-based discount.
+
+The CommitmentDiscountId column MUST be present in the billing data. This column must be of type String and MUST NOT contain null values when a charge is related to a commitment-based discount. When a charge is not associated with a commitment-based discount, the column MUST be null. CommitmentDiscountId MUST be unique within the provider.
```

Comment:
> when a charge is related to a commitment-based discount

Maybe: 'when a record is related to a commit-based discount'.

Indeed, in case of a resource covered by commitment-based discount, the charge (BilledCost) would be 0.
Maybe it could be explictly mentioned: the CommitDiscountId value must not be null if the record relates to the purchase of a commitment-based discount or if the record relates to the usage of a resource covered by a commitment-based discount.

Reply:
I'm fairly sure we've referred to the record / line-item / row as a 'charge' in other parts of the doc. Does a charge have to be a non-zero number? We need to update other places if we want to start referring to the line items as a record.

Examples are:
- ChargePeriod / ChargeType applies to all rows
- "Billing Currency is an identifier that represents the currency that a charge for resources and/or services was billed in."
- "If a provider does not support a sub account construct (only has a billing account) or does support a sub account construct, but the charge does not apply to a sub account, the SubAccountId column MUST be null."

Reply:
I don’t think a charge needs to be non-zero for this. I would still want to see that charge.

Reply:
Thanks for the clarification.
Indeed, I had not noticed that before a 'charge' was a synonym for a record / line-item / row.